frontend/test/simulate/utils.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2025 Microsoft Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simulate
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/mock/gomock"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/Azure/ARO-HCP/frontend/cmd"
+	"github.com/Azure/ARO-HCP/frontend/pkg/frontend"
+	"github.com/Azure/ARO-HCP/frontend/pkg/util"
+	"github.com/Azure/ARO-HCP/internal/audit"
+	"github.com/Azure/ARO-HCP/internal/database"
+	"github.com/Azure/ARO-HCP/internal/mocks"
+)
+
+type SimulationTestInfo struct {
+	CosmosDatabaseClient     *azcosmos.DatabaseClient
+	DBClient                 database.DBClient
+	MockClusterServiceClient *mocks.MockClusterServiceClientSpec
```

Comment:
nit: Could we define this as `ocm.ClusterServiceClientSpec` and not assume it's mocked?

I may want to leverage this framework for basic interaction tests w/ Cluster Service (in demo mode) in the interim before we drop it.